File: Ad-Hoc/pagina01/1030/1030.cpp
#include <iostream>
using namespace std;

int steps(int n, int k){
     if(n == 1) return 0;
    return (steps(n - 1, k) + k) % n;
}

int main () {

    int n, a, b;
    cin >> n;

    for(int i = 0; i < n; i++){

        cin >> a >> b;
        int step = steps(a, b);
        cout << "Case " + to_string(i + 1) + ": " + to_string(step + 1) << endl;
    }
    return 0;
}

File: Ad-Hoc/__1000__/1030/1030.php
<?php 

    function steps($n, $k){
        $result = 0;
        for($i = 2; $i < $n + 1; $i++){
            $result = ($result + $k) % $i;
        }
        return $result;
    }

    $n = (int) readline();
    for($i = 0; $i < $n; $i++){
        list($a,$b) = explode(" ", readline());
        $a = (int) $a;
        $b = (int) $b;
        $step = steps($a, $b) + 1;

        echo "Case " . ($i + 1) . ": $step\n";
    }
?>